libbeat/common/kubernetes/metadata.go

```diff
@@ -47,6 +49,9 @@ func (g *metaGenerator) PodMetadata(pod *Pod) common.MapStr {
 
 	annotationsMap := generateMapSubset(pod.Metadata.Annotations, g.annotations)
 	meta := common.MapStr{
+		"cluster": common.MapStr{
+			"name": g.cluster,
+		},
```

Comment:
Could we have it as a plain string like with `namespace`? I wonder why we have some names as strings and some others as `MapStr`.

Reply:
You mean `cluster.name: g.cluster` instead?  In general we always try to convert everything to already the nested object as otherwise it has to happen later or on the ES side.

Reply:
I mean not nesting :) `cluster: g.cluster` or `cluster_name: g.cluster`.

Reply:
@jsoriano if you look at GKE, there are lot of parameters that define a cluster and cluster name is just one of them. other information can be what is the region it is deployed against. so cluster being a MapStr allows placing cluster characteristics into the metadata.

Reply:
Ok then, agree, it is more flexible this way for the future.

Reply:
what vj said :-)